addon-test-support/@ember/test-helpers/settled.ts

```diff
@@ -44,20 +49,16 @@ if (typeof jQuery !== 'undefined' && _internalPendingRequests) {
   // This can be removed once Ember 4.0.0 is released
   EmberApplicationInstance.reopen({
     willDestroy(...args: any[]) {
-      const internalPendingRequests = _internalPendingRequests();
-
-      if (internalPendingRequests !== 0) {
```

Comment:
the outer if statement checks that `_internalPendingRequestsModule` exists, which I believe makes calling `off` and `clearPendingRequests` safe and desirable regardless of what the PendingRequestsCount is